drivers/pinctrl/common.c

```diff
@@ -9,6 +9,10 @@
 int pinctrl_lookup_state(const struct pinctrl_dev_config *config, uint8_t id,
 			 const struct pinctrl_state **state)
 {
+	if (config->state_cnt == 0U) {
```

Comment:
Wouldn't it be simpler to change the iteration below instead?

`while (*state < &config->states[config->state_cnt])`

Reply:
indeed, adjusted